src/Components/RecommendationsPage/RecommendationsTable.tsx

```diff
@@ -0,0 +1,111 @@
+import React from 'react';
+import { Button, Skeleton } from '@patternfly/react-core';
+import { Table, Tbody, Td, Th, Thead, ThProps, Tr } from '@patternfly/react-table';
+import { DateFormat } from '@redhat-cloud-services/frontend-components/DateFormat';
+
+import { RECOMMENDATIONS_DETAIL_ROOT } from '../../Helpers/constants';
+import { Recommendation } from './types';
+
+const RecommendationsTable = ({ recommendations, perPage, showSkeleton }) => {
+  const columnNames = {
+    description: 'Name',
+    publish_date: 'Modified',
+    systems: 'Systems',
+    remediation: 'Remediation',
+    run: 'Run',
+  };
+
+  const [activeSortIndex, setActiveSortIndex] = React.useState<number | null>(null);
+  const [activeSortDirection, setActiveSortDirection] = React.useState<'asc' | 'desc' | null>(null);
+
+  const getSortableRowValues = (rec: Recommendation): (string | number | Date)[] => {
+    const { description, publish_date, systems, remediation } = rec;
+    return [description, publish_date, systems, remediation];
+  };
+
+  if (activeSortIndex !== null) {
+    recommendations.sort((a, b) => {
+      const aValue = getSortableRowValues(a)[activeSortIndex];
+      const bValue = getSortableRowValues(b)[activeSortIndex];
+      if (typeof aValue === 'number') {
+        // Numeric sort
+        if (activeSortDirection === 'asc') {
+          return (aValue as number) - (bValue as number);
+        }
+        return (bValue as number) - (aValue as number);
+      } else {
+        // String sort
+        if (activeSortDirection === 'asc') {
+          return (aValue as string).localeCompare(bValue as string);
+        }
+        return (bValue as string).localeCompare(aValue as string);
+      }
+    });
+  }
```

Comment:
Create a new function in the helpers, so we can reuse it in future if we need. :+1: